lib/ae_mdw/db/sync/contract.ex

```diff
@@ -186,8 +168,73 @@ defmodule AeMdw.Db.Sync.Contract do
       {local_idx, "Channel.settle", :channel_settle_tx, _aetx, tx} ->
         Channels.settle_mutations({block_index, {call_txi, local_idx}}, tx)
 
+      {_local_idx, fname, :contract_create_tx, _aetx, tx} when fname in @contract_create_fnames ->
+        contract_pk = :aect_create_tx.contract_pubkey(tx)
+
+        [
+          aexn_create_contract_mutation(contract_pk, block_index, call_txi),
+          ContractCreateCacheMutation.new(contract_pk, call_txi)
+          | SyncOrigin.origin_mutations(
+              :contract_call_tx,
+              nil,
+              contract_pk,
+              call_txi,
+              tx_hash
+            )
+        ]
+
       {_local_idx, _fname, _tx_type, _aetx, _tx} ->
         []
     end)
   end
+
+  defp create_contract_create_aetx(block_hash, contract_pk, owner_id, initial_amount) do
+    {:ok, contract} = :aec_chain.get_contract(contract_pk)
+    deposit = :aect_contracts.deposit(contract)
+    abi_version = :aect_contracts.abi_version(contract)
+    vm_version = :aect_contracts.vm_version(contract)
+    code = retrieve_code(contract)
+    {_tag, owner_pk} = :aeser_id.specialize(owner_id)
+    owner_nonce = Db.nonce_at_block(block_hash, owner_pk)
+    nonce_tries = owner_nonce..(owner_nonce - 100)
```

Comment:
Backwards nonce happening would be a bit strange. Shouldn't it iterate forward in case two contracts are created (if possible)?

Reply:
Doesn't it get the nonce after the block execution, meaning to get the nonce from when the contract was executed we would need to go backwards

Reply:
The nonce is set by the user or sdk when submitting the transaction. So it is expected that the `contract_call_tx` gets the nonce set by the user and anything else that happens should increment it. 

Looked for a contract deployed this way and found this: https://testnet.aeternity.io/mdw/v2/txs/36393798

It raises an error after `owner_nonce - 100` become negative:
```
iex(aeternity@localhost)1> contract_pk = Validate.id!("ct_ZayzhtVmCTZB5SrN9viZ1gDortQAXnB7DzL2mFrYZBQ6G3SWE"); owner_pk = Validate.id!("ak_KXGNPxbw748m42xphh5uTGHHq85qKRvDtgHbHELLxfaPbDv72"); owner_nonce = AeMdw.Node.Db.nonce_at_block(Validate.id!("mh_2ifUzLrSH1Tsxk7VZZqfFmWSCo6PPSB1e7ogmiCyPJVbpHj43B"), owner_pk); nonce_tries = owner_nonce..(owner_nonce - 100); Enum.find(
...(aeternity@localhost)2>         nonce_tries,
...(aeternity@localhost)2>         &(:aect_contracts.compute_contract_pubkey(owner_pk, &1) == contract_pk)
...(aeternity@localhost)2>       )
** (FunctionClauseError) no function clause matching in :aect_contracts.compute_contract_pubkey/2    
    
    The following arguments were given to :aect_contracts.compute_contract_pubkey/2:
    
        # 1
        <<42, 14, 88, 87, 49, 171, 159, 216, 42, 105, 58, 14, 37, 87, 49, 119, 212, 21,
          89, 161, 213, 215, 206, 219, 148, 252, 51, 136, 114, 194, 186, 24>>
    
        # 2
        -1
```

Reply:
Interesting, does it find it going forwards?

Reply:
It doesn't work. The `sender_id` is found here: https://testnet.aeternity.io/mdw/v2/contracts/calls?contract=ct_KXGNPxbw748m42xphh5uTGHHq85qKRvDtgHbHELLxfaPbDv72